qiskit_nature/problems/second_quantization/lattice/lattice/triangular_lattice.py

```diff
@@ -0,0 +1,392 @@
+
+                node_a = rows * cols - 1
+                node_b = 0
+                graph.add_edge(node_b, node_a, np.conjugate(edge_parameter[2]))
+                self.boundary_edges.append((node_a, node_b))
+        else:
+            raise ValueError(
+                f"Invalid `boundary condition` {boundary_condition} is given."
+                "`boundary condition` must be `open` or `periodic`."
+            )
+
+        position_dict = {}
+        for index in range(np.prod(self.size)):
+            x = index % self.size[0]
+            y = index // self.size[0]
+            if self.boundary_condition == "open":
+                return_x = x
+                return_y = y
+            elif self.boundary_condition == "periodic":
+                return_x = x + 0.2 * np.sin(pi * y / (self.size[1] - 1))
+                return_y = y + 0.2 * np.sin(pi * x / (self.size[0] - 1))
+            position_dict[index] = [return_x, return_y]
+        self.position = position_dict
+        super().__init__(graph)
+
+    @classmethod
+    def from_adjacency_matrix(cls, input_adjacency_matrix: np.ndarray):
+        """Not implemented.
+
+        Args:
+            input_adjacency_matrix: Adjacency matrix with real or complex matrix elements.
+        Raises:
+            NotImplementedError
+        """
+        raise NotImplementedError()
+
+    # pylint: disable=arguments-differ
+    def draw(
+        self,
+        pos: Optional[dict] = None,
+        ax: Optional[Axes] = None,
+        arrows: bool = True,
+        arrowstyle: Optional[str] = None,
+        arrow_size: int = 10,
+        with_labels: bool = False,
+        node_list: Optional[list] = None,
+        edge_list: Optional[list] = None,
+        node_size: Union[int, list] = 300,
+        node_color: Union[
+            str,
+            Tuple[float, float, float],
+            Tuple[float, float, float],
+            List[Union[str, Tuple[float, float, float], Tuple[float, float, float, float]]],
+        ] = "#1f78b4",
+        node_shape: str = "o",
+        alpha: Optional[float] = None,
+        cmap: Optional[Colormap] = None,
+        vmin: Optional[float] = None,
+        vmax: Optional[float] = None,
+        linewidths: Union[float, Sequence] = 1.0,
+        width: Union[float, Sequence] = 1.0,
+        edge_color: Union[str, Sequence] = "k",
+        edge_cmap: Optional[Colormap] = None,
+        edge_vmin: Optional[float] = None,
+        edge_vmax: Optional[float] = None,
+        style: str = "solid",
+        labels: Optional[Callable] = None,
+        edge_labels: Optional[Callable] = None,
+        font_size: int = 12,
+        font_color: str = "k",
+        font_weight: str = "normal",
+        font_family: str = "sans-serif",
+        label: Optional[str] = None,
+        connectionstyle: str = "arc3",
+        self_loop: bool = False,
+        boundary_edges: bool = False,
+        **kwargs,
+    ):
+        r"""Draw the lattice.
```

Comment:
Same comments as before:
- do we need to duplicate the docstring?
- can we avoid duplicating all these kwargs?